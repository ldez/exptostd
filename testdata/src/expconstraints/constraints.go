package expconstraints

import (
	"time"

	"golang.org/x/exp/constraints" // want "Import statement 'golang.org/x/exp/constraints' can be replaced by 'cmp'"
)

type _[T constraints.Ordered] []T // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`

type _ interface {
	constraints.Ordered | time.Time // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
}

type _ interface {
	constraints.Ordered // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
	comparable
}

type _[K constraints.Ordered, V any] struct { // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
}

func _[T constraints.Ordered](_ []T) bool { // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
	return true
}

func _[S []E, E constraints.Ordered](s S) E { // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
	return s[len(s)-1]
}
