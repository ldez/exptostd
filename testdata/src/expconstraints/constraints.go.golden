package expconstraints

import (
	"time"

	"cmp" // want "Import statement 'golang.org/x/exp/constraints' can be replaced by 'cmp'"
)

type _[T cmp.Ordered] []T // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`

type _ interface {
	cmp.Ordered | time.Time // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
}

type _ interface {
	cmp.Ordered // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
	comparable
}

type _[K cmp.Ordered, V any] struct { // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
}

func _[T cmp.Ordered](_ []T) bool { // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
	return true
}

func _[S []E, E cmp.Ordered](s S) E { // want `golang.org/x/exp/constraints\.Ordered can be replaced by cmp\.Ordered`
	return s[len(s)-1]
}